pkgs/build-support/trivial-builders/default.nix

```
@@ -86,6 +86,10 @@ rec {
     , preferLocalBuild ? true
     , derivationArgs ? { }
     }:
+    assert lib.assertMsg (destination == "" || lib.hasPrefix "/" destination && destination != "/") ''
```

Could also be written with our logical implication operator, which appears to be slightly more common for such asserts.

Not blocking.

```suggestion
    assert lib.assertMsg (destination != "" -> (lib.hasPrefix "/" destination && destination != "/")) ''
```

Or, if we want to be really fancy:

```suggestion
    assert lib.assertMsg (destination != "" -> (lib.hasPrefix "/" destination)) ''
      destination must start with / if provided
    '';
    assert lib.assertMsg (destination != "/") ''
      destination cannot be /
    '';
```

---

I've kept this as one assert for now and updated the wording to make the `/` case a little more clear